src/main/java/org/jitsi/videobridge/health/Health.java

```diff
@@ -475,4 +476,110 @@ private static void interconnect(Endpoint[] endpoints)
         for (int i = 0; i < endpoints.length;)
             connect(endpoints[i++], endpoints[i++]);
     }
+
+    /**
+     * Starts a runnable which checks the health of {@code videobridge}
+     * periodically (at an interval of ({@link #HEALTH_CHECK_INTERVAL}).
+     * @param videobridge the {@link Videobridge} to run checks on.
+     */
+    public static void start(Videobridge videobridge)
+    {
+        if (checkers.get(videobridge) != null)
+        {
+            logger.error("Already running checks for " + videobridge);
+            return;
+        }
+
+        checkers.put(videobridge, new VideobridgePeriodicChecker(videobridge));
+    }
+
+    /**
+     * Stops running health checks for a specific {@link Videobridge}.
+     * @param videobridge the {@link Videobridge}.
+     */
+    public static void stop(Videobridge videobridge)
+    {
+        VideobridgePeriodicChecker checker = checkers.remove(videobridge);
+        if (checker != null)
+        {
+            checker.stop();
+        }
+    }
+
+    /**
+     * Periodically checks the health for a specific {@link Videobridge}
+     * instance and stores the results.
+     */
+    private static class VideobridgePeriodicChecker
+        extends PeriodicRunnableWithObject<Videobridge>
+    {
+        /**
+         * The exception resulting from the last health check performed on this
+         * videobridge. When the health check is successful, this is
+         * {@code null}.
+         */
+        private Exception lastResult = null;
+
+        /**
+         * The time the last health check finished being performed. A value of
+         * {@code -1} indicates that no health check has been performed yet.
+         */
+        private long lastResultMs = -1;
+
+        /**
+         * Initializes a new {@link VideobridgePeriodicChecker} instance for
+         * a specific {@link Videobridge} and registers this
+         * {@link PeriodicRunnable} with the executor.
+         *
+         * @param videobridge the {@link Videobridge}.
+         */
+        VideobridgePeriodicChecker(Videobridge videobridge)
+        {
+            super(videobridge, HEALTH_CHECK_INTERVAL);
+
+            executor.registerRecurringRunnable(this);
+        }
+
+        /**
+         * Performs a health check and updates this instance's state.
+         */
+        @Override
+        protected void doRun()
+        {
+            long start = System.currentTimeMillis();
+            Exception exception = null;
+
+            try
+            {
+                Health.doCheck(this.o);
+            }
+            catch (Exception e)
+            {
+                exception = e;
+            }
+
+            long duration = System.currentTimeMillis() - start;
+            lastResult = exception;
+            lastResultMs = start + duration;
+
+            if (exception == null)
+            {
+                logger.info(
```

Comment:
Wouldn't that be too chatty?

Reply:
It's just one line every 10 seconds, and I think it might be useful to have.